config/example.yaml

```diff
@@ -19,11 +22,53 @@ objects:
     - summoned/providera
     - prov/providera
     - org
-sparql:
-  endpoint: http://localhost/blazegraph/namespace/earthcube/sparql
-  endpointBulk: http://coreos.lan:3030/testing/data
-  endpointMethod: POST
-  contentType: application/n-quads
-  authenticate: false
-  username: ""
-  password: ""
+endpoints:
+  - server: blazegraph
```

Comment:
endpoints:
- ec_blazegraph:
   - baseurl: http://coreos.lan:9090/blazegraph/namespace/iow
   - type: blaszgraph
   - functions:
      - type: update
        path: /sparql
         content tyes:
         stuff2: 
     - type: bulk
        path: /sparql
- iow_graphdb
   - url http://coreos.lan:9090/blazegraph/namespace/iow/
   - type: graphed
   - functions 
      - type: update
         path: /sparql
         content tyes:
         stuff2: 
     - type: bulk
        url: /statements

Reply:
One issue is that for some triple stores the URL and METHOD is not always the same for SPARQL, SAPRQL UPDATE and "bulk loading".   Jena I know is that way as is oxigraph.

Reply:
baseurl + apl

So, as long as the starting path is the same, then this will be ok.
Otherwise do some abstractions, and inheritance to handle the quirks programatically?

Reply:
what do you think of this?  I still need to update the correct types and methods, just a structure demo

```yaml
endpoints:
  - service: ec_blazegraph
    baseurl: http://coreos.lan:9090/blazegraph/namespace/iow
    type: blaszgraph
    authenticate: false
    username: admin
    password: jfpwd
    modes:
      - mode: sparql
        suffix: /sparql
        contentType: text/x-nquads
        method: GET
      - mode: update
        suffix: /sparql
        contentType: text/x-nquads
        method: POST
      - mode: bulk
        suffix: /sparql
        contentType: text/x-nquads
        method: POST
  - service: iow_graphdb
    baseurl: http://coreos.lan:7200/repositories/testing
    type: graphed
    authenticate: false
    username: admin
    password: jfpw
    modes:
      - mode: sparql
        suffix: /
        contentType: text/x-nquads
        method: GET
      - mode: update
        suffix: /statements
        contentType: text/x-nquads
        method: POST
      - mode: bulk
        suffix: /statements
        contentType: text/x-nquads
        method: POST
```